inbm/dispatcher-agent/tests/unit/schedule/test_sqlite_manager.py

```diff
@@ -0,0 +1,104 @@
+from unittest import TestCase
+
+from dispatcher.schedule.sqlite_manager import SqliteManager
+from dispatcher.schedule.schedules import SingleSchedule, RepeatedSchedule
+from dispatcher.dispatcher_exception import DispatcherException
+
+class TestSqliteManager(TestCase):
+    
+    def setUp(self) -> None:
+        self.db = SqliteManager(":memory:")
+        
+    #def tearDown(self) -> None:
+        #if self.db._conn:
+        #    self.db._conn.close()
+     
+    def test_raise_exception_when_create_single_schedule_with_invalid_start_time(self):
+        ss1 = SingleSchedule(request_id="REQ123", 
+                            start_time=None,
+                            end_time="2024-01-02T00:00:00", 
+                            manifests=["MANIFEST1", "MANIFEST2"])
+        with self.assertRaisesRegex(DispatcherException, 
+                                    "Transaction failed: NOT NULL constraint failed: single_schedule.start_time"):
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Robustness$</h2>**
**Suggestion:** Use more descriptive exception messages that align with the operation being performed rather than generic database constraint messages.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        with self.assertRaisesRegex(DispatcherException, 
                                    "Invalid start time: Start time cannot be null."):
```